yt_dlp/extractor/bitchute.py

```diff
@@ -100,13 +106,18 @@ def _raise_if_restricted(self, webpage):
             self.raise_geo_restricted(reason)
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
+        base_url, video_id = self._match_valid_url(url).group('base_url', 'id')
         webpage = self._download_webpage(
             f'https://www.bitchute.com/video/{video_id}', video_id, headers=self._HEADERS)
 
         self._raise_if_restricted(webpage)
         publish_date = clean_html(get_element_by_class('video-publish-date', webpage))
         entries = self._parse_html5_media_entries(url, webpage, video_id)
+        details = get_element_by_class('details', webpage) or ''
+        uploader_path = extract_attributes(
+            get_element_html_by_class('spa', get_element_html_by_class('creator', details)) or '').get('href')
+        channel_path = extract_attributes(
+            get_element_html_by_class('spa', get_element_html_by_class('name', details)) or '').get('href')
```

Comment:
1. all of this "details" extraction should be moved down near the return for better code flow/readability
2. `uploader_path` and `channel_path` are currently fatal because `get_element_html_by_class` can return `None` and throws when its html input is `None`
3. matching `base_url` in `_VALID_URL` is overkill, we can always just use `https://www.bitchute.com`
4. use urljoin
5. move duplicated code into a function
```py
        details = get_element_by_class('details', webpage) or ''
        uploader_html = get_element_html_by_class('creator', details) or ''
        channel_html = get_element_html_by_class('name', details) or ''

        def construct_url(html):
            path = extract_attributes(get_element_html_by_class('spa', html) or '').get('href')
            return urljoin('https://www.bitchute.com', path)

        return {
            # ...
            'uploader': clean_html(uploader_html),
            'channel': clean_html(channel_html),
            'uploader_url': construct_url(uploader_html),
            'channel_url': construct_url(channel_html),
            # ...
        }
```